blackbox/docs/sql/general/value-expressions.rst

```diff
@@ -169,6 +169,34 @@ optionally prefixed with ``ARRAY``::
 
     :ref:`data-type-array-literals`
 
+.. _sql_expressions_array_subquery:
+
+Another way to construct an array is by using an ``ARRAY(subquery)`` expression
+as part of the :ref:`Select list <sql_reference_select_list>` of a ``SELECT`` statement::
+
+    ARRAY '(' subquery ')'
+
+Example::
+
+   cr> select mountain, height,
+   ... array(select height from sys.summits where country = 'AT' order by height desc limit 5) as array_top5_austrian_mountains
+   ... from sys.summits where country = 'AT' order by height desc limit 5;
+   +---------------------+--------+--------------------------------+
+   | mountain            | height | array_top5_austrian_mountains  |
+   +---------------------+--------+--------------------------------+
+   | Großglockner        |   3798 | [3798, 3770, 3666, 3564, 3550] |
+   | Wildspitze          |   3770 | [3798, 3770, 3666, 3564, 3550] |
+   | Großvenediger       |   3666 | [3798, 3770, 3666, 3564, 3550] |
+   | Großes Wiesbachhorn |   3564 | [3798, 3770, 3666, 3564, 3550] |
+   | Großer Ramolkogel   |   3550 | [3798, 3770, 3666, 3564, 3550] |
+   +---------------------+--------+--------------------------------+
+   SELECT 5 rows in set (... sec)
+
+
+.. NOTE::
+
+    Currently only subqueries returning one columns are supported for array constructors.
```

Comment:
I'd remove the `Currently`. It implies that this is a temporary limitation, which is not necessarily the case.